01-tree-specification.bs

```diff
@@ -134,19 +136,63 @@ This report also explains how clients MAY implement support for extracting conte
 
 Note: Having an identifier for the collection has become mandatory: without it you can otherwise not define completeness.
 
-# The member extraction algorithm # {#member-extraction-algorithm}
+# The Member Extraction Algorithm # {#member-extraction-algorithm}
+
+<div class="non-normative">
+RDF does not natively provide a way to reference a set of triples or quads, and can thus not unambiguously point to a member `M` without additional explanation.
+While [named graphs may be interpreted in a technology-specific stack as a reference to a set of triples](https://www.w3.org/TR/2014/NOTE-rdf11-datasets-20140225/), also other implementations exist that use it for different purposes.
+This makes it more complex: we both want to introduce a TREE-specific interpretation for named graph, as well as support existing datasets that already use named graphs for a different purpose.
+Therefore, this specification introduces a pragmatic set of decisions bundled in the Member Extraction Algorithm.
+Implementing this algorithm provides interoperability across clients that want to extract members in a common way.
+This way, it provides a way for a data provider to be clear about their intention towards a domain-agnostic processor.
+</div>
+
+Depending on its goals, a client MAY implement this algorithm to extract a set of quads that were intended as the member quads by the data provider.
+On the one hand, there is the TREE specific MEA that supports subject-based star patterns, named graphs and doing an extra HTTP request for out-of-band members.
+On the other hand, there is the generalized MEA that extends the TREE MEA with support for including more complex patterns, called shape topologies, and for partially out-of-band members.
+Additionally, there is also the TREE profile algorithm that provides a syntactic trick to get to lowest overhead possible for extracting members.
+
+Note: The MEA is indeed not mandatory for all TREE clients. For example, a client interested in autocompletion might be interested in only extracting the text literals. A client focused on SPARQL querying will know what quads it wants to select. However, a client that is built to domain agnostically do an operation on top of the TREE collection and pass it on to a next step, might want to understand the full package of quads that was intended by the data provider.
+
+## The TREE Member Extraction Algorithm ## {#tree-member-extraction-algorithm}
+
+The TREE MEA is a combination of subject-based star patterns (cfr. [[!CBD]]) and named graphs.
+First we introduce a couple of symbols:
+ * `N` is the set of all named graphs used in the current page, including the default graph.
+ * `Nb` is the set of blank nodes in `N`, excluding the default graph.
+ * `f` is the first focus node of a member we are extracting found in a `<yourcollection> tree:member ?f` triple.
```

Comment:
The "first" in "first focus node" was confusing me a bit. Can one member have multiple focus nodes? The second part of the sentence makes more sense to me: the [single] focus node of a member is identified by `?f` in the triple `<yourcollection> tree:member ?f`.